sherpa/astro/tests/test_astro_data2.py

```diff
@@ -1111,6 +1111,58 @@ def test_pha_invalid_units(invalid, make_test_pha):
     assert str(de.value) == f"unknown quantity: '{invalid}'"
 
 
+# currently several variants are allowed, even if they don't work as
+# expected - e.g. type="rates" does not set the rate value to True.
```

Comment:
Wait, I thought this PR fixes that so that the variants that don't work as expected no longer work?

Reply:
Did I mess something up in my rebasing? These tests shouldn't be marked as xfail (and the comment above should be removed). Let me check

Reply:
My rebase left these in when they can be removed (ie the xfail tests suddenly started passsing). So I've made the change in "remove un-needed xfail markers".